packages/babel-plugin-transform-react-jsx/src/create-plugin.ts

```diff
@@ -31,11 +31,13 @@ const DEFAULT = {
   pragmaFrag: "React.Fragment",
 };
 
-const JSX_SOURCE_ANNOTATION_REGEX = /\*?\s*@jsxImportSource\s+([^\s]+)/;
-const JSX_RUNTIME_ANNOTATION_REGEX = /\*?\s*@jsxRuntime\s+([^\s]+)/;
+const JSX_SOURCE_ANNOTATION_REGEX =
+  /^\*?\s*?\*?\*?\s*@jsxImportSource\s+([^\s]+)\s*$/;
```

Comment:
I think we have too many `*`s: we shouldn't accept `* ** @jsxImportSource` (`\s*?\*?` should be deleted from the regexp)

Reply:
Oh I just saw https://github.com/babel/babel/issues/13896#issuecomment-982985687. I think it can be correctly handled using a regexp like
```js
const SPACE_NO_NEWLINE = /(?:(?![\r\n])\s)/.source;
const JSX_SOURCE_ANNOTATION_REGEX = new RegExp(
  String.raw`${SPACE_NO_NEWLINE}*\*?${SPACE_NO_NEWLINE}*@jsxImportSource\s+([^\s]+)[\s\*]*$`,
  "m"
);
```

Or even just
```js
const JSX_SOURCE_ANNOTATION_REGEX = /\s*\*?\s*@jsxImportSource\s+([^\s]+)[\s\*]*$/m;
```

even if it accepts too much newlines.

Reply:
Okay, I will remove the extra spaces.